### PR TITLE
Squashed grok commits

### PR DESCRIPTION
**Pull request by cdaconta**
Removed EmbeddingProvider from class

Removed EmbeddingProvider from class

Added to config Grok and created GrokProvider.java

Changed class name on GrokProvider and added Grok config details on config.properties.template

Added import to XlmEcosystemServiceImpl

Addeed GrokProvider to META-INF

Changed supportsEmbeddings to false

Adding config.properties.template back.

### CODE REVIEW

**Comment by cdaconta** [comment]
I didn't mean to close it.  whoops

**Comment by mdaconta** [comment]
Excellent! I will review...

File: src/main/java/us/daconta/xlmeco/GrpcXlmServer.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ private static Properties loadProperties(String fileName) throws IOException {
 
     public static void main(String[] args) throws Exception {
         // Load properties from config file
-        Properties properties = loadProperties("config.properties");
+        Properties properties = loadProperties("config.properties.template");
```
Comment (mdaconta) [inline src/main/java/us/daconta/xlmeco/GrpcXlmServer.java:29]: You don't load the template.  Need to change this back to config.properties.  The template is only used to copy it to the config.properties file and add your API keys.  I actually have my "real" file stored away in another directory and I just copy it in when I want to test.

File: src/main/java/us/daconta/xlmeco/provider/impl/GrokProvider.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import us.daconta.xlmeco.provider.GenerativeProvider;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+public class GrokProvider extends AbstractGenerativeProvider implements ChatProvider, EmbeddingProvider {
```
Comment (mdaconta) [inline src/main/java/us/daconta/xlmeco/provider/impl/GrokProvider.java:21]: Why does it implement EmbeddingProvider if it cannot provide embeddings?  Recommend removing that and setting the supports "embedding" property to false.

File: src/main/java/us/daconta/xlmeco/provider/impl/GrokProvider.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                    ChatResponsePart responsePart = ChatResponsePart.newBuilder().setToken(token).build();
+                    responseObserver.onNext(responsePart);
+                }
+            }
+        }
+        responseObserver.onCompleted();
+    }
+    @Override
+    public List<Float> generateEmbedding(String text, ModelParameters params) {
+        // Create the JSON body for the request
```
Comment (mdaconta) [inline src/main/java/us/daconta/xlmeco/provider/impl/GrokProvider.java:157]: Is this from the OpenAI embedding?  I thought Grok did not support this...

**Comment by mdaconta** [comment]
The file config.properties.template should not have been deleted.  Please return that file to the repo.

**Review by mdaconta — CHANGES_REQUESTED** [review]
A few minor issues with this.  Please correct these and resubmit the PR.

**Comment by cdaconta** [comment]
I made the changes you stated.

**Review by mdaconta — APPROVED** [review]

**Comment by mdaconta** [comment]
merged!

**Pull request merged**